### ISSUE TITLE
0.17.0 regression: named imports of commonjs lib raise error

### ISSUE THREAD

**Issue by nusgnaf** [antfu/vite-ssg#150]
**Env**
Node v16.13.0
nite-ssg 0.17.0

While playing with vitesse templates, I try third party UI components, after `pnpm i -S naive-ui`, I import component by add one liner inside `pages/index.vue`
`import {NButton} from naive-ui`
while `npm run build` raise the following error:

```
[vite-ssg] An internal error occurred.
[vite-ssg] Please report an issue, if none already exists: https://github.com/antfu/vite-ssg/issues
file:///home/fang/Project/web/vitesse-main/.vite-ssg-temp/main.mjs:27
import { NButton } from "naive-ui";
         ^^^^^^^
SyntaxError: Named export 'NButton' not found. The requested module 'naive-ui' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:
```
vite-ssg 0.16.2 give green light on `npm run build`

**Comment by moritzruth** [comment]
I encountered the same issue with https://www.npmjs.com/package/blobs

**Comment by antfu** [comment]
Try set

```ts
{
  ssgOptions: {
    format: 'cjs'
  }
}
```

in your vite.config.js with v0.17.2

**Comment by AshenBird** [comment]
I have similar problem and I have update to v0.17.2. But the problem has occur. If I set config
``` javascript
{
  ssgOptions: {
    format: 'cjs'
  }
}
```
the other error will appear. 
Like this:
``` shell
Error: Cannot find module 'file:\D:\path\to\vscode-notebook\src\client\explorer\.vite-ssg-temp\index.cjs'
Require stack:
- D:\path\to\vscode-notebook\node_modules\.pnpm\registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d\node_modules\vite-ssg\dist\node\cli.mjs
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at build (file:///D:/path/to/vscode-notebook/node_modules/.pnpm/registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d/node_modules/vite-ssg/dist/node/cli.mjs:185:77)
    at async Object.handler (file:///D:/path/to/vscode-notebook/node_modules/.pnpm/registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d/node_modules/vite-ssg/dist/node/cli.mjs:298:3) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\path\\to\\vscode-notebook\\node_modules\\.pnpm\\registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d\\node_modules\\vite-ssg\\dist\\node\\cli.mjs'
  ]
}
```
[My repository](https://github.com/AshenBird/vscode-notebook)
build commad : `npm run build:client`

**Comment by userquin** [comment]
@AshenBird It is not a regresion is windows file system, I'll take a look...

**Comment by AshenBird** [comment]
> Try set
> 
> ```ts
> {
>   ssgOptions: {
>     format: 'cjs'
>   }
> }
> ```
> 
> in your vite.config.js with v0.17.2

@userquin I means this config can't fix this error. It will cause another error.

like this.

> I have similar problem and I have update to v0.17.2. But the problem has occur. If I set config
> 
> ```js
> {
>   ssgOptions: {
>     format: 'cjs'
>   }
> }
> ```
> 
> the other error will appear. Like this:
> 
> ```shell
> Error: Cannot find module 'file:\D:\path\to\vscode-notebook\src\client\explorer\.vite-ssg-temp\index.cjs'
> Require stack:
> - D:\path\to\vscode-notebook\node_modules\.pnpm\registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d\node_modules\vite-ssg\dist\node\cli.mjs
>     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
>     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
>     at Module.require (node:internal/modules/cjs/loader:1005:19)
>     at require (node:internal/modules/cjs/helpers:102:18)
>     at build (file:///D:/path/to/vscode-notebook/node_modules/.pnpm/registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d/node_modules/vite-ssg/dist/node/cli.mjs:185:77)
>     at async Object.handler (file:///D:/path/to/vscode-notebook/node_modules/.pnpm/registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d/node_modules/vite-ssg/dist/node/cli.mjs:298:3) {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: [
>     'D:\\path\\to\\vscode-notebook\\node_modules\\.pnpm\\registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d\\node_modules\\vite-ssg\\dist\\node\\cli.mjs'
>   ]
> }
> ```
> 
> [My repository](https://github.com/AshenBird/vscode-notebook) build commad : `npm run build:client`

**Comment by sibbng** [comment]
@AshenBird Seems related to changes in #153

@userquin If it helps similar fix at peeky https://github.com/Akryum/peeky/commit/34826cbd1a2c3beba65b13de44cbda6532c6512a

**Comment by AshenBird** [comment]
@sibbng I have update to `v0.17.2` , but it not work. 
``` shell
Error: Cannot find module 'file:\D:\path\to\vscode-notebook\src\client\explorer\.vite-ssg-temp\index.cjs'
Require stack:
- D:\path\to\vscode-notebook\node_modules\.pnpm\registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d\node_modules\vite-ssg\dist\node\cli.mjs
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at build (file:///D:/path/to/vscode-notebook/node_modules/.pnpm/registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d/node_modules/vite-ssg/dist/node/cli.mjs:185:77)
    at async Object.handler (file:///D:/path/to/vscode-notebook/node_modules/.pnpm/registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d/node_modules/vite-ssg/dist/node/cli.mjs:298:3) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\path\\to\\vscode-notebook\\node_modules\\.pnpm\\registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d\\node_modules\\vite-ssg\\dist\\node\\cli.mjs'
  ]
}
```
And I set config like

> Try set
> 
> ```ts
> {
>   ssgOptions: {
>     format: 'cjs'
>   }
> }
> ```
> 
> in your vite.config.js with v0.17.2

then 

``` shell
Error: Cannot find module 'file:\D:\path\to\vscode-notebook\src\client\explorer\.vite-ssg-temp\index.cjs'
Require stack:
- D:\path\to\vscode-notebook\node_modules\.pnpm\registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d\node_modules\vite-ssg\dist\node\cli.mjs
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at build (file:///D:/path/to/vscode-notebook/node_modules/.pnpm/registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d/node_modules/vite-ssg/dist/node/cli.mjs:185:77)
    at async Object.handler (file:///D:/path/to/vscode-notebook/node_modules/.pnpm/registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d/node_modules/vite-ssg/dist/node/cli.mjs:298:3) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\path\\to\\vscode-notebook\\node_modules\\.pnpm\\registry.npmmirror.com+vite-ssg@0.17.2_6fabc582b63c5c1e05ffcfb709d04e5d\\node_modules\\vite-ssg\\dist\\node\\cli.mjs'
  ]
}
```
But `v0.17.2` has fixed #153 in theory. I cant't confirm, they are same problem.

**Comment by leosin** [comment]
same proplem !!
Isn't this what evan recommended?
Are there any other options besides cjs?
Because I also need to use swiper,it only supports esm mode 😮‍💨
it's really hard to choose😭

**Comment by userquin** [comment]
> @AshenBird Seems related to changes in #153
> 
> @userquin If it helps similar fix at peeky [Akryum/peeky@34826cb](https://github.com/Akryum/peeky/commit/34826cbd1a2c3beba65b13de44cbda6532c6512a)

I've been patching `esm` resolution on windows almost 1 month on `vitest`, the line involved here should use the following function (I'll check it later):

```ts
// https://github.com/vitest-dev/vitest/blob/77ba9fa8894cc9328676ec8bf729ec192986db37/packages/vite-node/src/externalize.ts#L103
function patchWindowsImportPath(path: string) {
  if (path.match(/^\w:\\/))
    return `file:///${slash(path)}`
  else if (path.match(/^\w:\//))
    return `file:///${path}`
  else
    return path
}
```

**Comment by yongchin0821** [comment]
I meet a seem propblem
Please report an issue, if none already exists: https://github.com/antfu/vite-ssg/issues
(node:3575) UnhandledPromiseRejectionWarning: file:///Users/yangqing/Workspace/DEV/myblog/dev-ssg/.vite-ssg-temp/main.mjs:2
import { NIcon, NMenu, NGrid, NGridItem } from "naive-ui";
                       ^^^^^
SyntaxError: Named export 'NGrid' not found. The requested module 'naive-ui' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'naive-ui';
const { NIcon, NMenu, NGrid, NGridItem } = pkg;

    at ModuleJob._instantiate (internal/modules/esm/module_job.js:121:21)
    at async ModuleJob.run (internal/modules/esm/module_job.js:166:5)
    at async Loader.import (internal/modules/esm/loader.js:178:24)
    at async Object.build (/Users/yangqing/Workspace/DEV/myblog/dev-ssg/node_modules/vite-ssg/dist/chunks/build.cjs:171:87)
    at async Object.handler (/Users/yangqing/Workspace/DEV/myblog/dev-ssg/node_modules/vite-ssg/dist/node/cli.cjs:25:3)
(Use `node --trace-warnings ...` to show where the warning was created)



I tried to set 
{
  ssgOptions: {
    format: 'cjs'
  }
}

it didn't work

**Comment by userquin** [comment]
@nusgnaf @yongchin0821 I'm moving the CLI to ESM and should fix the problem when using `format: 'cjs'`, I have [AshenBird/vscode-notebook](https://github.com/AshenBird/vscode-notebook) working just removing the `const vscode = acquireVsCodeApi();` missing function, using `format: 'cjs'` (PR in a few minutes, we need to review it):

https://streamable.com/in7mco

**Comment by userquin** [comment]
@yongchin0821 can you test it with this? Just remove the `vite-ssg` dep and add it from file (if you're using `pnpm`, just drop the `zip` file uncompress it and copy the `tgz` (I cannot upload hete `tgz` files)  on your root directory, then run `pnpm remove -D vite-ssg && pnpm add -D ./vite-ssg-0.19.2.tgz`):


[vite-ssg-0.19.2.tgz.zip](https://github.com/antfu/vite-ssg/files/8597025/vite-ssg-0.19.2.tgz.zip)

**Comment by yongchin0821** [comment]
@userquin 
env:macos 12.1

1. yarn remove vite-ssg
2. yarn add -D ~/vite-ssg-0.19.2.tgz
3. yarn build
```shell
dev-ssg git:(dev-ssg) ✗ yarn build   
yarn run v1.22.10
$ vite-ssg build
env: node\r: No such file or directory
error Command failed with exit code 127.
```

**Comment by userquin** [comment]
uhmm, can you provide a repo?

**Comment by userquin** [comment]
@nusgnaf Try with this one:

[vite-ssg-0.19.2.tgz.zip](https://github.com/antfu/vite-ssg/files/8597092/vite-ssg-0.19.2.tgz.zip)

**Comment by yongchin0821** [comment]
> uhmm, can you provide a repo?

well,sorry my **experimental [project](https://github.com/yongchin0821/testrep/tree/main/dev-ssg)**,it's going to refactoring

**Comment by userquin** [comment]
@yongchin0821 i'll take a look in a few minutes, whatching svelte summit... ;)

**Comment by userquin** [comment]
@yongchin0821 using `pnpm`, and once installed all `markdown-it**` dependencies, removing `highlight.js/styles/monokai.css` from `src/components/Readmd.vue` and  `src/layouts/body.vue` and adding this to `vite.config.ts` file just works:
```ts
  ssgOptions: {
    mock: true,
    format: 'cjs',
  }
```

![imagen](https://user-images.githubusercontent.com/6311119/166111822-12f5bd03-6592-4a29-9d4c-fa43a5652407.png)

![imagen](https://user-images.githubusercontent.com/6311119/166111861-8ad547cf-8199-43e0-b2f0-a2c05ff6f558.png)

**Comment by userquin** [comment]
@yongchin0821 just check my fork here https://github.com/userquin/vite-ssg-esm-cli-150, check my commit https://github.com/userquin/vite-ssg-esm-cli-150/commit/8159c7cc61e5d3a75082fab5e049079b8134789c

**Comment by userquin** [comment]
> @AshenBird Seems related to changes in #153
> 
> @userquin If it helps similar fix at peeky [Akryum/peeky@34826cb](https://github.com/Akryum/peeky/commit/34826cbd1a2c3beba65b13de44cbda6532c6512a)

the new PR should fix it
